windows-gaming-driver/src/driver/mod.rs

```diff
@@ -96,10 +97,22 @@ pub fn run(cfg: &Config, tmp: &Path, data: &Path) {
         Box::new(catch_sigterm),
     ]).map(|_| ());
 
-    match core.run(qemu.select(joined)) {
-        Ok(((), _)) => (), // one (*hopefully* qemu) is done, so the other is too
-        Err((e, _)) => panic!("Unexpected error: {}", e),
-    }
-
+    core.run(qemu.join(joined.or_else(|x| { error!("Unexpected error: {}", x); Ok(()) }))).expect("Unexpected error");
+
+	info!("unbinding resetable vfio-things");
+	
+    for slot in cfg.machine.vfio_slots.iter() {
+		if let &VfioDevice::Temporarily(ref device) = slot {
+			let mut child = Command::new(Path::new(::DATA_FOLDER).join("vfio-ubind")).arg(device).arg("-r").spawn().expect("failed to run vfio-ubind");
+			match child.wait() {
+				Ok(status) => 
+						if !status.success() {	
+						error!("vfio-ubind failed with {}! The device might still be bound to the vfio-driver!", status);
+						},
+				Err(err) => error!("failed to wait on child. Got: {}", err)
+				
+			}
+		}
+	}
```

Comment:
>The chance that we panic is pretty high.

Is it? If so, why? Note that this was restructured to no longer panic when an error is returned.